### ISSUE TITLE
Exception occurred: Error: Remote 'g' packet reply is too long

### ISSUE THREAD

**Issue by vani5hing** [pwndbg/pwndbg#2573]
### Description
- Error while use gdb.debug() a pwn challenge
**https://github.com/guyinatuxedo/nightmare/tree/master/modules/30-unlink/hitcon14_stkof**
(I will provide a binary later, just in case)
- Please help, I dont want to met this error again during a contest
- Here is error screenshot
![Screenshot 2024-11-26 084404](https://github.com/user-attachments/assets/52895799-4313-4221-a6c5-28d9a8020fa5)

### Steps to reproduce
- Create a new folder then put the **stkof** and **libc-2.23.so** file to there then type **pwninit** , it will create all essential file to do a challenge, include **stkof_patched**
- Type **python3 ex.py** in the command shell to run the script (I will provide the **ex.py** (my scripts) file)
- When the debug shell appear, as I already set the breakpoint, just follow Gdb session history:
```
c
c
c
ni
```

### My setup
Platform: Linux-5.15.167.4-microsoft-standard-WSL2-x86_64-with-glibc2.40
OS: Kali GNU/Linux Rolling
OS ABI: #1 SMP Tue Nov 5 00:21:55 UTC 2024
Architecture: x86_64
Endian: little
Charset: utf-8
Gdb:      15.2
Python:   3.12.7 (main, Nov  8 2024, 17:55:36) [GCC 14.2.0]
Pwndbg:   2024.02.14 build: 1ea126a0
Capstone: 5.0.1280
Unicorn:  2.0.1

### Provided file
[files.zip](https://github.com/user-attachments/files/17911387/files.zip)

**Comment by patryk4815** [comment]
Before executing to new binary, you need delete breakpoints 🙈 

```
pwndbg> info breakpoint
Num     Type           Disp Enb Address            What
1       breakpoint     keep y   0x0000000000400c1c
        breakpoint already hit 1 time
2       breakpoint     keep y   0x0000000000400b7a
        breakpoint already hit 2 times
pwndbg> d 2
pwndbg> d 1
pwndbg> ni

<ctrl+c   after 10s>


pwndbg> procinfo
exe        '/bin/sh'
cmdline    /bin/sh ''
cwd        '/work'
pid        9078
tid        9078
selinux    kernel
ppid       9077
uid        [0, 0, 0, 0]
gid        [0, 0, 0, 0]
groups     [0]
fd[0]      pipe:[6982538]
fd[1]      /dev/pts/2
fd[2]      /dev/pts/2
```

**Comment by patryk4815** [comment]
this looks like bug in gdb https://github.com/pwndbg/pwndbg/issues/1754 , But maybe errors could be improved
```

  File "/work/pwndbg/lib/python3.12/site-packages/pwndbg/gdblib/hooks.py", line 26, in update_typeinfo
    lang = gdb.execute("show language", to_string=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gdb.error: Remote 'g' packet reply is too long (expected 560 bytes, got 2420 bytes): daffffffffffffff00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000090b9ffffff7f00000

  File "/work/pwndbg/lib/python3.12/site-packages/pwndbg/gdblib/hooks.py", line 26, in update_typeinfo
    lang = gdb.execute("show language", to_string=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gdb.error: Remote 'g' packet reply is too long (expected 560 bytes, got 2420 bytes): daffffffffffffff00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000090b9ffffff7f00000

  File "/work/pwndbg/lib/python3.12/site-packages/pwndbg/dbg/gdb.py", line 604, in arch
    if "little" in gdb.execute("show endian", to_string=True).lower():
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gdb.error: Remote 'g' packet reply is too long (expected 560 bytes, got 2420 bytes): daffffffffffffff000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a0b9ffffff7f00000
```

**Comment by disconnect3d** [comment]
Should be fixed by the workaround implemented in #2573 (see also https://github.com/pwndbg/pwndbg/issues/2576).

**Comment by k4lizen** [comment]
@disconnect3d you probably mean "workaround implemented in https://github.com/pwndbg/pwndbg/pull/2577" :p
